system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -991,8 +991,24 @@ class CLibXpathContext : public CInterfaceOf<IXpathContext>, implements IVariabl
         switch (obj->type)
         {
             case XPATH_NODESET:
-                toXMLNodeSet(obj, content);
-                isValue = false;
+                if (isValue)
+                {
+                    if (obj->nodesetval->nodeNr > 1)
+                    {
+                        isValue = false;
+                        result = false;
```

Comment:
If result is false you have no idea if isValue=false is relevant.  For example the function may have returned at line 985 above, and you'd not know if isValue was updated or not.

Reply:
The original intent of isValue was to distinguish between nodeset and other evaluation results. The thinking at the time was that all values would be passed to maskValue and all node sets would be passed to maskContent. This was wrong, since an XPath to get an element value evaluates to a nodeset. The value is now predominantly an input parameter, remaining a reference for exactly one scenario - a value is expected and multiple nodes are found.

`trace-content` disregards isValue on output.

`trace-value` is only concerned with isValue on output when result is false. If this method returns at line 985, isValue will still be true and `trace-value` will not register a warning for node ambiguity. Returning false at line 1024 will trigger a warning if isValue has changed.

If this method could be counted on to register an appropriate warning, isValue could be changed from reference to value. I don't think it would be appropriate to pass a trace messenger from esdllib to xmllib to get the message we want.

Reply:
I supose it's fine.  If you get a return of false and isValue is changed you can tell it was a nodeset above 1.  It's the kind of routine the caller has to understand internals of which isn't ideal but it's fine.